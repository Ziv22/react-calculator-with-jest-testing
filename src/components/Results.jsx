const Results = ({result}) => {
    return (
      <div className="result" data-testid="result"> {result} </div>
    );
  }
  
export default Results;