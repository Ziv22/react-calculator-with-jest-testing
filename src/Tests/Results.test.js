import React from "react";
import ReactDOM from "react-dom";
import Results from "../components/Results";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

const result = 7;

test("Renders without crashing", () => {
  const root = document.createElement("div");
  ReactDOM.render(<Results result={result} />, root);
});

test("Renders Results with correct data", () => {
  const { getByTestId } = render(<Results result={result} />);
  expect(getByTestId("result")).toHaveTextContent(result);
});

test("Result component matches snapshot", () => {
  const resultSnapshot = renderer.create(<Results result={result} />).toJSON();
  expect(resultSnapshot).toMatchSnapshot();
});
