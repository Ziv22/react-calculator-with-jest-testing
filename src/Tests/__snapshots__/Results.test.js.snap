// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component matches snapshot 1`] = `
<div
  className="result"
  data-testid="result"
>
   
  7
   
</div>
`;
